### PR TITLE
feat: Parser veeam

### PR DESCRIPTION
**Pull request by cwadhwani-splunk**
feat: Parser veeam

### CODE REVIEW

File: package/etc/conf.d/conflib/syslog/app-syslog-veeam_mp.conf [hunk 1]
```diff
@@ -0,0 +1,18 @@
+block parser app-syslog-veeam_mp() {
+    channel {
+        rewrite {
+            r_set_splunk_dest_default(
+                index("infraops")
+                sourcetype('veeam_vbr_syslog')
```
Comment (ikheifets-splunk) [inline package/etc/conf.d/conflib/syslog/app-syslog-veeam_mp.conf:6]: we ususally defining sourcetype as `vendor:product`

File: docs/sources/vendor/Veeam/veeam.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+## Key facts
+
+* MSG Format based filter
+* rfc5424 default port 514
+
+## Links
+
+| Ref            | Link                                                                                                    |
+|----------------|---------------------------------------------------------------------------------------------------------|
+| Splunk Add-on | <https://splunkbase.splunk.com/app/7312/>                                                              |
```
Comment (ikheifets-splunk) [inline docs/sources/vendor/Veeam/veeam.md:12]: it's developer supported add-on probably we don't need to mention it on documentation for SC4S. @rjha-splunk @mstopa-splunk

**Review by ikheifets-splunk — APPROVED** [review]

File: docs/sources/vendor/Veeam/veeam.md (hunk 2 above)
Comment (rjha-splunk) [reply docs/sources/vendor/Veeam/veeam.md:12]: We will remove this section

File: package/etc/conf.d/conflib/syslog/app-syslog-veeam_mp.conf [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+                vendor('veeam')
+                product('veeam')
```
Comment (rjha-splunk) [inline package/etc/conf.d/conflib/syslog/app-syslog-veeam_mp.conf:8]: product('vbr')

Comment (rjha-splunk) [reply package/etc/conf.d/conflib/syslog/app-syslog-veeam_mp.conf:8]: add class field here
class('syslog')

File: package/etc/conf.d/conflib/syslog/app-syslog-veeam_mp.conf (hunk 1 above)
Comment (rjha-splunk) [inline package/etc/conf.d/conflib/syslog/app-syslog-veeam_mp.conf:6]: sourcetype('veeam:vbr:syslog')

File: package/lite/etc/addons/veeam/app-syslog-veeam_mp.conf [hunk 4]
```diff
@@ -0,0 +1,18 @@
+block parser app-syslog-veeam_mp() {
+    channel {
+        rewrite {
+            r_set_splunk_dest_default(
+                index("infraops")
+                sourcetype('veeam_vbr_syslog')
```
Comment (rjha-splunk) [inline package/lite/etc/addons/veeam/app-syslog-veeam_mp.conf:6]: same as core

File: tests/test_veeam.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    mt = env.from_string(
+        """{{ mark }} {{ iso }} {{ host }} Veeam_MP - - [origin enterpriseId="3xxx3"] [categoryId=0 instanceId=10010 OibID="1a583aa0-84f0-4f63-8cc0-a2e25a3dxxxf" OriginalOibID="75ae4bc8-725b-4583-b8e6-c2dfcabxxxba" VmRef="d5692942-d615-d64b-2339-31e3c29xxx49" VmName="server.mydomain.com" ServerName="This server" DateTime="06/10/2024 14:32:17" IsCorrupted="True" Platform="6" StorageSize="28473884672" RepositoryID="1451444b-83fa-44ec-9965-a48dxxx954cd" IsFull="False" Version="1" Description="VM 'server.mydomain.com' restore point has been created."]"""
+    )
+    message = mt.render(mark="<14>1", host=host, bsd=bsd, iso=iso)
+
+    sendsingle(message, setup_sc4s[0], setup_sc4s[1][514])
+
+    st = env.from_string(
+        'search _time={{ epoch }} index=infraops host="{{ host }}" sourcetype="veeam_vbr_syslog"'
```
Comment (rjha-splunk) [inline tests/test_veeam.py:32]: sourcetype="veeam:vbr:syslog"'

File: docs/sources/vendor/Veeam/veeam.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+| User Guide | <https://helpcenter.veeam.com/docs/backup/vsphere/overview.html?ver=120>                                                              |
+## Sourcetypes
+
+| sourcetype     | notes                                                                                                   |
+|----------------|---------------------------------------------------------------------------------------------------------|
+| veeam_vbr_syslog        | sourcetype decided considering the Veeam Splunk Add-on                                                                                               |
```
Comment (rjha-splunk) [inline docs/sources/vendor/Veeam/veeam.md:18]: mention 
key
veeam_vbr_syslog
Sourcetype
veeam:vbr:syslog

**Review by rjha-splunk — COMMENTED** [review]
Please address all comments

**Review by rjha-splunk — APPROVED** [review]

**Pull request merged**